test/e2e/storage/testsuites/snapshottable.go

```diff
@@ -79,167 +87,320 @@ func (s *snapshottableTestSuite) SkipRedundantSuite(driver TestDriver, pattern t
+		framework.Logf("deleting snapshot %q/%q", l.vs.GetNamespace(), l.vs.GetName())
+		// typically this snapshot has already been deleted
+		err = l.dc.Resource(SnapshotGVR).Namespace(l.vs.GetNamespace()).Delete(context.TODO(), l.vs.GetName(), metav1.DeleteOptions{})
+		if err != nil && !apierrors.IsNotFound(err) {
+			framework.Failf("Error deleting snapshot %q. Error: %v", l.pvc.Name, err)
+		}
+	})
+	err = WaitForSnapshotReady(l.dc, l.vs.GetNamespace(), l.vs.GetName(), framework.Poll, framework.SnapshotCreateTimeout)
+	framework.ExpectNoError(err)
+
+	ginkgo.By("checking the snapshot")
+	// Get new copy of the snapshot
+	l.vs, err = l.dc.Resource(SnapshotGVR).Namespace(l.vs.GetNamespace()).Get(context.TODO(), l.vs.GetName(), metav1.GetOptions{})
+	framework.ExpectNoError(err)
+
+	// Get the bound snapshotContent
+	snapshotStatus := l.vs.Object["status"].(map[string]interface{})
+	snapshotContentName := snapshotStatus["boundVolumeSnapshotContentName"].(string)
+	l.vscontent, err = l.dc.Resource(SnapshotContentGVR).Get(context.TODO(), snapshotContentName, metav1.GetOptions{})
+	framework.ExpectNoError(err)
+
+	snapshotContentSpec := l.vscontent.Object["spec"].(map[string]interface{})
+	volumeSnapshotRef := snapshotContentSpec["volumeSnapshotRef"].(map[string]interface{})
+
+	// Check SnapshotContent properties
+	ginkgo.By("checking the SnapshotContent")
+	framework.ExpectEqual(snapshotContentSpec["volumeSnapshotClassName"], l.vsc.GetName())
+	framework.ExpectEqual(volumeSnapshotRef["name"], l.vs.GetName())
+	framework.ExpectEqual(volumeSnapshotRef["namespace"], l.vs.GetNamespace())
+}
+
+// TestSnapshotDeleted tests the results of deleting a VolumeSnapshot
+// depending on the deletion policy currently set.
+func TestSnapshotDeleted(l *snapshottableLocal, sct SnapshotClassTest) {
+	var err error
+	snapshotStatus := l.vs.Object["status"].(map[string]interface{})
+	snapshotContentName := snapshotStatus["boundVolumeSnapshotContentName"].(string)
+
+	ginkgo.By("deleting the snapshot")
+	err = l.dc.Resource(SnapshotGVR).Namespace(l.vs.GetNamespace()).Delete(context.TODO(), l.vs.GetName(), metav1.DeleteOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		framework.Failf("Error deleting snapshot %s in namespace %s. Error: %v", l.vs.GetName(), l.vs.GetNamespace(), err)
+	}
+
+	ginkgo.By("checking the Snapshot has been deleted")
+	err = WaitForGVRDeletion(l.dc, SnapshotGVR, l.sc.Name, framework.Poll, framework.SnapshotDeleteTimeout)
+	framework.ExpectNoError(err)
+
+	if sct.DeletionPolicy == "Delete" {
+		ginkgo.By("checking the SnapshotContent has been deleted")
+		err = WaitForGVRDeletion(l.dc, SnapshotContentGVR, snapshotContentName, framework.Poll, framework.SnapshotDeleteTimeout)
+		framework.ExpectNoError(err)
+	} else if sct.DeletionPolicy == "Retain" {
+		ginkgo.By("checking the SnapshotContent has not been deleted")
+		err = WaitForGVRDeletion(l.dc, SnapshotContentGVR, snapshotContentName, 1*time.Second, 10*time.Second)
```

Comment:
wonder if 10 secs too less a time ensure that snapshot deletion is not triggered? @msau42 thoughts?

Reply:
I'll up to it to 30 seconds, but I don't want to pause the test too long for this. I'm open to other durations for this.